agent/consul/discoverychain/gateway.go

```diff
@@ -128,6 +128,29 @@ func (l *GatewayChainSynthesizer) Synthesize(chains ...*structs.CompiledDiscover
 			return nil, nil, err
 		}
 
+		node := compiled.Nodes[compiled.StartNode]
```

Comment:
The problematic router is always/only the first entry in the chain?

Reply:
Yes, in fact you can only have a single router in the entire chain since routers can only route to splitters/resolvers, and splitters can only route to splitters/resolvers -- so the only place you can use a router is in the very first part of a discovery chain.

Regardless though, we only ever create a single router as the first node in our synthetic chain.